### ISSUE TITLE
Recover Backend After Amplify Delete

### ISSUE THREAD

**Issue by kimfucious** [aws-amplify/amplify-cli#3381]
**Amplify CLI Version** 
4.13.1

**What AWS Services are you utilizing?**
Auth(Cognito), API Gateway (REST), DynamoDB, Lambda, & S3

**Details**
Okay, this one is for the stupid-user-record-book.  So while I was futzing around trying to get a react native app working with an existing back-end, I *imported* (not a pull) an environment into the react native app per [these steps](https://github.com/aws-samples/create-react-app-auth-amplify/blob/master/README.md#run-locally-with-the-amplify-cli).

In short, this resulted in a `team-provider-info.json` file being created with a the awscloudformation data, representing the env that was imported.

It didn't actually work the way I expected it to, so I ran `amplify delete`, thinking it would simply clean up what I just did my running `amplify init`.  Not smart.

Unfortunately, because of the `team-provider-info.json` entry that I'd just imported, this basically wiped out the entire infrastructure in the cloud.   Oops 🙄 

So, currently, the app is up and running via Cloudfront without a back-end, and I need to put it back.

I do have everything backed up, including the dynamoDB data.  There's been no changes to the original amplify code (in the app where it was originally provisioned).  And the repo is also unchanged.

At present an `amplify status` shows no changes, because (I think) there's no diffs between the `backend` and the `#current-cloud-backend` directories.

I've read a couple of suggestions on how to recover from this, but I'm asking here for (hopefully) some form of "official" and--better yet-- a magical, step-by-step --answer, that will save me the trouble of re-building the back-end manually.

**Comment by kimfucious** [comment]
**Progress update:**

I decided to bit the bullet and try the following:

**Amplify recovery steps:**

1. Backup amplify directory (somewhere else)
2. Delete existing amplify directory
3. Run `amplify init` in project root
4. Answer prompts
5. This will create on AWS:
    * On AWS:
      - a cloudformation stack
      - an s3 bucket (for deployment)
      - 2 IAM roles (auth and unAuth)
    * In the root dir of the app:
      - Amplify directory
6. Recreate all services with `amplify cli`.  Use amplify/backend in backed up dir from step one above to remind yourself.
    * Auth
      - Recreate users
      - Adjust policies, settings, etc.
    * API
      - Paths
      - Functions
        - Add npm packages in function src dir
        - Copy old function code from backup into each function
      - Dynamo Storage
        - Change Capacity to On-Demand (if desired)
      - S3 Storage (for object storage)
7. Amplify Push
    - Values in `aws-exports` are the old values!

**Current Status:**
After replacing all of the values in `aws-exports` manually (I know this is a bad idea) with the new, updated values, I can access the newly provisioned back-end when running the app locally.

However, as expected, deployments via the AWS console are broken because somewhere the old `aws-exports` values are hiding, and they are getting pulled when the build starts.

**Additional Question**
How can I force the `aws-exports` values to get updated, when a `amplify push` doesn't do the trick?

**Comment by ammarkarachi** [comment]
@kimfucious I am assuming you didn't get a chance to change the aws-export.js before pushing the first time? have you tried changing the values and then pushing for the first time. Because for the consequent pushes it does a merge

**Comment by kimfucious** [comment]
Thanks, @ammarkarachi 

Sorry for the delay in responding, I got sidetracked on something else.

I believe that your assumption is correct, though I don't actually remember.

What I wound up doing was making a change to one of the lambda functions in the re-created back-end and then pushing once more.  This resulted in the an update to the `aws-export.js` values on my local. 

This also created a new app in the Amplify Console containing the new back-end, where I had to reset all the settings, like: env vars, redirects, custom domain, etc.

I'm thinking maybe I could have imported the back-end into the existing app, but deleting everything had me a bit gun-shy with the import process.  Maybe next time 😉
